### PR TITLE
Adding more complicated todo mvc

### PR DESCRIPTION
**Pull request by rorticus**
A version of `todo-mvc` but with more features!

* Adding searching
* Adding todo detail page
* Adding card view

### CODE REVIEW

File: todo-mvc-kitchensink/package.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "dojo-interfaces": ">=2.0.0-alpha.8",
+    "grunt": "^1.0.1",
+    "grunt-tslint": "^3.0.0",
+    "grunt-dojo2": "^2.0.0-beta.16",
+    "intern": "~3.4.1",
+    "tslint": "^3.15.1",
+    "typescript": "~2.1.4"
+  },
+  "dependencies": {
+    "@reactivex/rxjs": "5.0.0-beta.6",
```
Comment (dylans) [inline todo-mvc-kitchensink/package.json]: Is this needed, shouldn't it use the observables shim instead?

File: todo-mvc-kitchensink/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		{ name: 'dojo-compose', location: 'node_modules/dojo-compose' },
+		{ name: 'dojo-routing', location: 'node_modules/dojo-routing' },
+		{ name: 'dojo-core', location: 'node_modules/dojo-core' },
+		{ name: 'dojo-shim', location: 'node_modules/dojo-shim' },
+		{ name: 'dojo-has', location: 'node_modules/dojo-has' },
+		{ name: 'dojo-stores', location: 'node_modules/dojo-stores' },
+		{ name: 'dojo-widgets', location: 'node_modules/dojo-widgets' },
+		{ name: 'immutable', location: 'node_modules/immutable/dist', main: 'immutable' },
+		{ name: 'maquette', location: 'node_modules/maquette/dist', main: 'maquette' },
+		{ name: 'rxjs', location: 'node_modules/@reactivex/rxjs/dist/amd' }
```
Comment (dylans) [inline todo-mvc-kitchensink/src/index.ts]: Is this needed, shouldn't it use the observables shim instead?

File: todo-mvc-kitchensink/tests/intern.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		{ name: 'dojo-core', location: 'node_modules/dojo-core' },
+		{ name: 'dojo-dom', location: 'node_modules/dojo-dom' },
+		{ name: 'dojo-has', location: 'node_modules/dojo-has' },
+		{ name: 'dojo-routing', location: 'node_modules/dojo-routing' },
+		{ name: 'dojo-shim', location: 'node_modules/dojo-shim' },
+		{ name: 'dojo-stores', location: 'node_modules/dojo-stores' },
+		{ name: 'dojo-widgets', location: 'node_modules/dojo-widgets' },
+		{ name: 'immutable', location: 'node_modules/immutable/dist', main: 'immutable' },
+		{ name: 'maquette', location: 'node_modules/maquette/dist', main: 'maquette' },
+		{ name: 'rxjs', location: 'node_modules/@reactivex/rxjs/dist/amd' }
```
Comment (dylans) [inline todo-mvc-kitchensink/tests/intern.ts]: Is this needed, shouldn't it use the observables shim instead?

File: todo-mvc-kitchensink/typings.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+	"name": "dojo2-todo-mvc",
+	"globalDependencies": {
+		"extra-rxjs": "github:dojo/typings/custom/dojo2-extras/rxjs.d.ts#103096ce945dd51a18cc47a9b7cf9191fdac0349",
```
Comment (dylans) [inline todo-mvc-kitchensink/typings.json]: Is this needed, shouldn't it use the observables shim instead?

File: todo-mvc-kitchensink/package.json (hunk 1 above)
Comment (rorticus) [inline todo-mvc-kitchensink/package.json]: Good catch! Removed all these pesky rxjs references.

File: todo-mvc-kitchensink/LICENSE [hunk 5]
```diff
@@ -0,0 +1,28 @@
+The "New" BSD License
+*********************
+
+Copyright (c) 2015 - 2016, The Dojo Foundation
```
Comment (dylans) [inline todo-mvc-kitchensink/LICENSE]: Should be 2017, JS Foundation

Comment (rorticus) [inline todo-mvc-kitchensink/LICENSE]: Updated!

File: todo-mvc-kitchensink/package.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (dylans) [inline todo-mvc-kitchensink/package.json]: It's probably time to change these to scoped dependencies (as the ci is now failing)... though it looks like it's complaining about the usage of the Observable shim now.

Comment (rorticus) [inline todo-mvc-kitchensink/package.json]: Updated to `@dojo` namespace and the latest widget dependencies

File: todo-mvc-kitchensink/package.json [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  "name": "dojo2-todo-mvc",
+  "version": "0.0.1",
+  "description": "TodoMVC using Dojo2",
```
Comment (dylans) [inline todo-mvc-kitchensink/package.json]: minor nit, Dojo 2 rather than Dojo2

File: todo-mvc-kitchensink/package.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  "dependencies": {
+    "@dojo/compose": "^2.0.0-beta.21",
+    "@dojo/core": "^2.0.0-alpha.20",
+    "@dojo/has": "^2.0.0-alpha.7",
+    "@dojo/loader": "^2.0.0-beta.9",
+    "@dojo/routing": "^2.0.0-alpha.8",
+    "@dojo/shim": "^2.0.0-beta.8",
+    "@dojo/stores": "^2.0.0-alpha.9",
+    "@dojo/widgets": "^2.0.0-alpha.19",
+    "immutable": "^3.8.1",
```
Comment (dylans) [inline todo-mvc-kitchensink/package.json]: Do we still need immutable?

File: todo-mvc-kitchensink/readme.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
+# Dojo2 TodoMVC Example
```
Comment (dylans) [inline todo-mvc-kitchensink/readme.md]: Missed this before, let's standardize on Dojo 2.

File: todo-mvc-kitchensink/readme.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+> Dojo2 TodoMVC example.
```
Comment (dylans) [inline todo-mvc-kitchensink/readme.md]: Dojo 2

File: todo-mvc-kitchensink/readme.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+> Current progress of Dojo2 is available from the _[Meta](https://github.com/dojo/meta)_ repository.
```
Comment (dylans) [inline todo-mvc-kitchensink/readme.md]: Dojo 2

**Comment by rorticus** [comment]
OK @dylans , those issues should all be taken care of!

**Pull request merged**